src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,145 @@
     `${event.block.block.header.number.toNumber()}-${event.idx}`
   );
+
   uniqueTransfer.block = event.block.block.header.number.toNumber();
   uniqueTransfer.from = from.toString();
   uniqueTransfer.to = to.toString();
-  uniqueTransfer.collectionId = collectionId.toString();
-  uniqueTransfer.itemId = itemId.toString();
   if (event.extrinsic) {
+    event.extrinsic.events.forEach((eventItem) => {
+      if (eventItem.event.method === "Withdraw") {
+        const fee = eventItem.event.data[1];
+        uniqueTransfer.fee = (fee as Balance).toBigInt();
+      }
+    });
     uniqueTransfer.txHash = event.extrinsic.extrinsic.hash.toString();
     uniqueTransfer.timestamp = new Date(
-    event.extrinsic.block.timestamp
-  ).getTime();
+      event.extrinsic.block.timestamp
+    ).getTime();
   }
 
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.owner = to.toString();
+  uniqueTransfer.itemId = item.id;
+  uniqueTransfer.collectionId = collection.id;
+
+  await collection.save();
+  await item.save()
   return uniqueTransfer.save();
 }
+
+export const handleUniquesMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug("uniqueMetadataSetEvent added: " + JSON.stringify(event.toHuman()));
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const data = event.event.data[2];
+  // const isFrozen = event.event.data[3];
+
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.metadataCid = data.toHuman().toString();
+  item.collectionId = collection.id;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCollectionMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "uniqueCollectionMetadataSetEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const data = event.event.data[1];
+  // const isFrozen = event.event.data[2];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.metadataCid = data.toHuman().toString();
+
+  return collection.save();
+}
+
+export const handleUniquesDestroyedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesDestroyedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.isDestroyed = true;
+  return collection.save()
+}
+
+export const handleUniquesBurnedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesBurnedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const itemId = event.event.data[1];
+
+  const item = await Item.get(itemId.toString());
+  item.isBurned = true;
+  return item.save();
```

Comment:
Could it be better to:
```suggestion
  return Item.remove(itemId.toString());
```

Reply:
Not really, since the items and collections have relation with transaction history so on delete it could cause an rule violation
We need the transaction history to be persistent

Reply:
Same here. It seems to me that the item id is reusable after and item is burnt. Then in querying the history, the item id would not be enough to link transactions to correct items. Thus internally and on the subquery side,  we need to make a unique id for items for example based on the block number they created in. Or better than that based on the transaction hash that created them.